mwclient/entity.py

```diff
@@ -0,0 +1,271 @@
+"""WikiBase Entities and related objects."""
+
+
+class Entity(object):
+
+    """Wikibase Entity, either Item or Property.
+
+    This class should not be implemented directly,
+    It is meant to be abstract for Item and Property.
+
+    Attributes:
+        site (WikiBaseSite): reference to a WikiBaseSite
+        entity (str): Q number of the entity.
+        descriptions (dict): dictionnary containing description per language
+        labels (dict): dictionnary containing labels per language
+    """
+
+    def __init__(self, site, normalized_entity):
+        """Common part of constructor for Item and Property."""
+        self.site = site
+
+        self.entity = normalized_entity
+        # caching descriptions, labels, sitelinks
+        # self._sitelinks = None
+        self._descriptions = None
+        self._labels = None
+
+        # caching claims
+        self.itemclaims = None
+
+    def setinfofromwbgetentities(self, result):
+        """Set descriptions, labels and claims from wbgetentities result."""
+        self._descriptions = dict()
+        for language in result['descriptions']:
+            lang = result['descriptions'][language]['language']
+            value = result['descriptions'][language]['value']
+            self._descriptions[lang] = value
+        self._labels = dict()
+        for language in result['labels']:
+            lang = result['labels'][language]['language']
+            value = result['labels'][language]['value']
+            self._labels[lang] = value
+        if self.itemclaims is None:
+            self.itemclaims = []
+            for prop in result['claims']:
+                for claim in result['claims'][prop]:
+                    mainsnak = claim['mainsnak']
+                    self.itemclaims.append(Claim.fromsnak(self.site, mainsnak))
+
+    @property
+    def labels(self):
+        """Labels dictionnary per language"""
+        if self._labels is None:
+            entities = self.site.api('wbgetentities', ids=self.entity)
+            result = entities['entities'][self.entity]
+            self.setinfofromwbgetentities(result)
+        return self._labels
+
+    @property
+    def descriptions(self):
+        """Descriptions dictionnary per language"""
```

Comment:
"Get a dictio<s>n</s>nary of descriptions per language code"

Reply:
fixed!